### PR TITLE
docs: add `IngressClassName` to `ApisixRoute` spec

### LINKED ISSUE #2201: docs: is IngressClassName supported in ApisixRoute custom resource?
### Current State

Is the IngressClassName supported in `ApisixRoute` and `ApisixUpstream`? The following URL has not any mention of it anywhere,
https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2/

However, after searching I found that this is closed as completed: https://github.com/apache/apisix-ingress-controller/issues/1089 , following the merge of this Pull Request that says it adds support for it: https://github.com/apache/apisix-ingress-controller/pull/593 

### Desired State

There should be a mention in the specification documentation about the support of `IngressClassName`, if it is actually supported.

### PR DESCRIPTION
**Pull request by kayx23**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [x] Documentation
- [ ] Refactor
- [ ] Chore
- [ ] CI/CD or Tests

### What this PR does / why we need it:

Closes #2201 

Add `IngressClassName` to `ApisixRoute` spec.

### Pre-submission checklist:

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: docs/en/latest/references/apisix_route_v2beta3.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ The table below describes each of the attributes in the spec. The fields `apiVer
 
 | Attribute                            | Type               | Description                                                                                                                                                                                 |
 |--------------------------------------|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ingressClassName                                 | string              | Name of the ingress class. Used to target a specific controller that should implement the resource.                                                                                                                                                                         |
```
Comment (kayx23) [inline docs/en/latest/references/apisix_route_v2beta3.md:38]: Unsure if `v2beta3` API needs this update.

**Review by pottekkat — COMMENTED** [review]
LGTM.

You don't need to update the v2beta3 doc. In fact, I think we can remove these "beta" API docs entirely, as they are no longer needed.

cc: @Revolyssup

**Comment by kayx23** [comment]
> You don't need to update the v2beta3 doc. In fact, I think we can remove these "beta" API docs entirely, as they are no longer needed.

I thought so too seeing some previous discussions but need confirmation.

**Comment by pottekkat** [comment]
@kayx23 I checked with @Revolyssup. We can remove these as the v2beta3 APIs are no longer in the CRDs.

**Comment by kayx23** [comment]
I reverted the change for `v2beta3` in this PR. Removing docs for `v2beta3` will go in a different PR.

**Review by pottekkat — APPROVED** [review]
Sounds good. LGTM.

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**